### PR TITLE
[script.service.checkpreviousepisode@matrix] 0.4.1

### PR DESCRIPTION
**Pull request by bossanova808**
### Add-on details:

- **General**
  - Add-on name: Kodi Check Previous Episode
  - Add-on ID: script.service.checkpreviousepisode
  - Version number: 0.4.1
  - Kodi/repository version: matrix

- **Code location**
  - URL: https://github.com/bossanova808/script.service.checkpreviousepisode
  
This service helps prevent spoliers by checking if the previous episode in a series has been watched. If not, it will pause playback and warn you.  Specific shows can be marked to be ignored if episode order does not matter.

### Checklist:

- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-scripts/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

### CODE REVIEW

**Comment by bossanova808** [comment]
@Razzeee Here's the result of your work/magic - thanks!

File: script.service.checkpreviousepisode/addon.xml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  <extension library="default.py" point="xbmc.service" />
+  <extension point="xbmc.addon.metadata">
+    <summary lang="en_GB">Kodi Check Previous Episode</summary>
+    <summary lang="bg_BG">Изгледан ли е предходния епизод</summary>
+    <description lang="en_GB">This service helps prevent spoilers by checking if the previous episode in a series has been watched. If not, it will pause playback and warn you.  Specific shows can be marked to be ignored if episode order does not matter.</description>
+    <description lang="de_DE">Dieses Script überprüft die vorige Episode der TV Show die gerade gestartet werden soll und warnt falls diese noch nicht gesehen wurde oder nicht vorhanden ist!</description>
+    <description lang="bg_BG">Когато пускате епизод от даден сериал скриптът проверява предходния и предупреждава ако той отсъства от библиотеката или не е изгледан.</description>
+    <platform>all</platform>
+    <language />
+    <license>GNU GENERAL PUBLIC LICENSE. Version 2, June 1991</license>
```
Comment (enen92) [inline script.service.checkpreviousepisode/addon.xml:17]: spdx if possible

File: script.service.checkpreviousepisode/addon.xml [hunk 2]
```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<addon id="script.service.checkpreviousepisode" name="Kodi Check Previous Episode" provider-name="bossanova808, Razzeee, Lucleonhart" version="0.4.1">
+  <requires>
+    <import addon="xbmc.python" version="3.0.0" />
+    <import addon="script.module.yaml" version="3.11.0" />
+	<import addon="xbmc.addon" version="18.9.701" />
```
Comment (enen92) [inline script.service.checkpreviousepisode/addon.xml:6]: please remove this dependency, no longer required

File: script.service.checkpreviousepisode/addon.xml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  </requires>
+  <extension library="default.py" point="xbmc.service" />
+  <extension point="xbmc.addon.metadata">
+    <summary lang="en_GB">Kodi Check Previous Episode</summary>
+    <summary lang="bg_BG">Изгледан ли е предходния епизод</summary>
+    <description lang="en_GB">This service helps prevent spoilers by checking if the previous episode in a series has been watched. If not, it will pause playback and warn you.  Specific shows can be marked to be ignored if episode order does not matter.</description>
+    <description lang="de_DE">Dieses Script überprüft die vorige Episode der TV Show die gerade gestartet werden soll und warnt falls diese noch nicht gesehen wurde oder nicht vorhanden ist!</description>
+    <description lang="bg_BG">Когато пускате епизод от даден сериал скриптът проверява предходния и предупреждава ако той отсъства от библиотеката или не е изгледан.</description>
+    <platform>all</platform>
+    <language />
```
Comment (enen92) [inline script.service.checkpreviousepisode/addon.xml:16]: drop as it does not provide any content

File: script.service.checkpreviousepisode/addon.xml [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    <license>GNU GENERAL PUBLIC LICENSE. Version 2, June 1991</license>
+    <forum>https://forum.kodi.tv/showthread.php?tid=151531</forum>
+    <website>https://kodi.wiki/view/Add-on:XBMC_Check_Previous_Episode</website>
+    <source>https://github.com/bossanova808/script.service.checkpreviousepisode</source>
+  <assets>
+     <icon>icon.png</icon>
+  </assets>
+  </extension>
+</addon>
```
Comment (enen92) [inline script.service.checkpreviousepisode/addon.xml:25]: missing new line at the end of the file (cosmetic)

File: script.service.checkpreviousepisode/default.py [hunk 5]
```diff
@@ -0,0 +1,242 @@
...
+import yaml
+from distutils.util import strtobool
+
+from traceback import format_exc
+
+__addon__ = xbmcaddon.Addon()
+__cwd__ = __addon__.getAddonInfo('path')
+__scriptname__ = __addon__.getAddonInfo('name')
+__version__ = __addon__.getAddonInfo('version')
+__kodiversion__ = float(xbmcaddon.Addon('xbmc.addon').getAddonInfo('version')[0:4])
```
Comment (enen92) [inline script.service.checkpreviousepisode/default.py:16]: you can also use System infolabels to get the kodi version

File: script.service.checkpreviousepisode/addon.xml (hunk 1 above)
Comment (Razzeee) [reply script.service.checkpreviousepisode/addon.xml:17]: You can find the valid SPDX strings here https://spdx.org/licenses/

**Comment by bossanova808** [comment]
Ok this one should be good to go.  Thanks again :)

**Comment by bossanova808** [comment]
Sorry, rookie error.  have pushed an update to correct the typo.

**Pull request merged**